six/modules/c++/six.sicd/include/six/sicd/Grid.h

```diff
@@ -127,6 +141,42 @@ struct DirectionParameters
         return !(*this == rhs);
     }
 
+    bool validate(const ImageData& imageData,
+            logging::Logger& log) const;
+    bool validate(const RgAzComp& rgAzComp,
+            logging::Logger& log,
+            double offset = 0) const;
+
+    void fillDerivedFields(const ImageData& imageData);
+    void fillDerivedFields(const RgAzComp& rgAzComp, const GeoData& geoData,
+            double offset = 0);
+private:
+    std::auto_ptr<Functor> calculateWeightFunction() const;
+
+    bool validateWeights(const Functor& weightFunction,
+            logging::Logger& log) const;
+
+    double derivedKCenter(const RgAzComp& rgAzComp,
+            double offset = 0) const;
+
+    Poly2D derivedKcoaPoly(const RgAzComp& rgAzComp,
+            double offset = 0) const;
+
+    std::vector<RowColInt>
+            calculateImageVertices(const ImageData& imageData) const;
+
+    /* Return vector contents, in order:
+    * 0) deltaK1 (min)
+    * 1) deltaK2 (max)
+    */
+    std::pair<double, double> calculateDeltaKs(
+            const ImageData& imageData) const;
+
+    const double WGT_TOL = 1e-3;
+    const size_t DEFAULT_WEIGHT_SIZE = 512;
+    const std::string boundsErrorMessage =

```

Comment:
Here and throughout, this should be all caps with underscores.  Also change it to a `const char []` to be slightly more efficient unless it makes usage more of a pain.